### PR TITLE
Fix wiki

### PR DESCRIPTION
**Pull request by HThuren**
Part of solution to #3521 
Change the logic to get the right Wiki for board on flasher tab.

Logic to find right url are:
1. if board description exist in https://betaflight.com/docs/wiki/boards/ + target name
2. if no board description found, show board missing = https://betaflight.com/docs/wiki/boards/missing

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by HThuren** [comment]
@blckmn @haslinghuis 
compare of bords files .md and unified target files .config looks like we need to rename the boardfiles to same convention than unified target files, ie. KAKUTEH7.md -> HBRO-KAKUTEH7.md
Will require a change to this PR

What do you think ?

**Comment by haslinghuis** [comment]
Firmware 4.5 has only the board name in the config file.

Probably focus on 4.5 first. Board name should be used as wiki page (file name) on betaflight.com - so we might have to rename some - didn't check).

**Comment by HThuren** [comment]
> Firmware 4.5 has only the board name in the config file.
> 
> Probably focus on 4.5 first. Board name should be used as wiki page (file name) on betaflight.com - so we might have to rename some - didn't check).

OK, I understand you mean - no rename. Then the PR are are ready to merge.

PS. we have 138 board documentation and 376 unified target files

File: src/js/tabs/firmware_flasher.js
Comment (haslinghuis) [inline src/js/tabs/firmware_flasher.js:119]:
```suggestion
            if (urlExists(urlBoard)) {
                urlWiki = urlBoard;
            }
```

File: src/js/utils/common.js
Comment (haslinghuis) [inline src/js/utils/common.js:99]:
```suggestion
    if (http.status !== 404) {
```

File: src/js/tabs/firmware_flasher.js
Comment (HThuren) [reply src/js/tabs/firmware_flasher.js:119]: I think we should keep the overide possibility, we can ue this to point more config files to one single documentaion, if needed
(I will stop working for today now)

Comment (haslinghuis) [reply src/js/tabs/firmware_flasher.js:119]: The url does not work now with KAKUTEH7 without this change.

Rather we use target name and default path in documentation and remove the dependency on unified targets repo reducing maintenance otherwise we need to update config.h too.

File: src/js/utils/common.js
Comment (haslinghuis) [inline src/js/utils/common.js:103]: image

```suggestion
    return http.status !== 404;
```

File: src/js/tabs/firmware_flasher.js
Comment (HThuren) [reply src/js/tabs/firmware_flasher.js:119]: Only KAKUTEF4 has the right #wiki string , the others are from before betaflight website was redesigned.

My suggestion are like you want, a default to (2.) if board description exist in https://betaflight.com/docs/wiki/boards/ + target name
But still keep the override possibility, can solve fx if we want to point to documentation in archive for old boards, and we already have the logic in summary struct

Comment (haslinghuis) [reply src/js/tabs/firmware_flasher.js:119]: The other reason we should not use summary.wiki is that firmware 4.5 does not use unified targets configuration.
As we already have the target board name itself there is no need to explicit use a configuration file for a pointer to documentation. Much easier to create 4 pages for the variants [for the ALIEANFLIGHTF7] and be done.

Comment (HThuren) [reply src/js/tabs/firmware_flasher.js:119]: OK, now I get it, wiki way obsolute in firmware 4.5

**Review by haslinghuis — APPROVED** [review]

**Comment by nerdCopter** [comment]
"https://betaflight.com/docs/wiki/boards/missing" is a 404 page; is this what is desired?

**Comment by haslinghuis** [comment]
@nerdCopter yes see: https://github.com/betaflight/betaflight.com/pull/207 :grimacing:

**Review by nerdCopter — APPROVED** [review]

**Pull request merged**
